spmd/tensor/ops/tensor_ops.py

```diff
@@ -53,3 +63,163 @@ def no_shard_prop_rule(op_schema: OpSchema) -> OutputSharding:
 
 for op in no_shard_prop_ops:
     DTensor._op_to_rules[op] = no_shard_prop_rule
+
+
+def unshard_tensor_dim(
+    placements: Sequence[Placement], dim: int
+) -> Sequence[Placement]:
+    """Disallow the given tensor dimension to be sharded"""
+    return tuple(
+        p if (not isinstance(p, Shard) or p.dim != dim) else Replicate()
+        for p in placements
+    )
+
+
+def _prop_all_but_dim(
+    op_schema: OpSchema, dim: int, out_shape: torch.Size
+) -> OutputSharding:
+    """
+    Considering an op that takes its input as first argument, forwards all shardings
+    except for the given dimension.
+    """
+    input_spec = op_schema.args_schema[0]
+    assert isinstance(input_spec, DTensorSpec)
+
+    output_placements = unshard_tensor_dim(input_spec.placements, dim=dim)
+    output_spec = DTensorSpec(
+        mesh=input_spec.mesh,
+        placements=output_placements,
+        shape=out_shape,
+        ndim=input_spec.ndim,
+    )
+
+    if input_spec.placements == output_placements:
+        out = OutputSharding(output_spec=output_spec)
+    else:
+        suggested_input_spec = DTensorSpec(
+            mesh=input_spec.mesh,
+            placements=output_placements,
+            ndim=input_spec.ndim,
+            shape=input_spec.shape,
+        )
+        out = OutputSharding(
+            output_spec=None,
+            schema_suggestions=[
+                OpSchema(
+                    args_schema=(suggested_input_spec,)
+                    + op_schema.args_schema[1:],
+                    kwargs_schema=op_schema.kwargs_schema,
+                ),
+            ],
+        )
+    return out
+
+
+@register_prop_rule("aten.slice.Tensor")
+def prop_slice(op_schema: OpSchema) -> OutputSharding:
+    """NOTE: can be further optimized (right now it replicates before slicing on a sharded dimension)"""
+    defaults = (None, 0, None, None, 1)
+    input_spec, dim, start, end, step = (
+        op_schema.args_schema + defaults[len(op_schema.args_schema) :]
+    )
+    assert isinstance(input_spec, DTensorSpec)
+    assert isinstance(dim, int)
+    assert start is None or isinstance(start, int)
+    assert end is None or isinstance(end, int)
+    assert isinstance(step, int)
+
+    # normalize arguments
+    if dim < 0:
+        dim += input_spec.ndim
+    if start is None:
+        start = 0
+    if step is None:
+        step = 1
+    if end is None or end > input_spec.shape[dim]:
+        end = input_spec.shape[dim]
+    if end < 0:
+        end += input_spec.shape[dim]
+    if end > input_spec.shape[dim]:
```

Comment:
Is this already handled by line 138 or is this a catch for line 140's condition?

Reply:
@anj-s good point, we can remove this.